### PR TITLE
2242 - Added OWASP Maven Dependency checker

### PR DESCRIPTION
**Pull request by Akrish17**
[2242 - Added OWASP Maven Dependency checker]

#### 📲 What

Added OWASP Maven Dependency checker in the pom.xml

#### 🤔 Why

To detect publicly disclosed vulnerabilities contained within a project’s maven dependencies.

#### 🛠 How

OWASP will determine if there is a Common Platform Enumeration (CPE) identifier for a given dependency. If found, it will generate a report linking to the associated CVE entries.
Need to run `mvn clean install -Powasp-dependency-check` as part of each build. This will generate a dependency-check-report.html under the target folder. 

#### 👀 Evidence
image

#### 🕵️ How to test

Notes for QA

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

**Comment by ElvenSpellmaker** [comment]
Does this need anything to hook into the build?

**Comment by Akrish17** [comment]
> Does this need anything to hook into the build?

Yes

File: java/pom.xml [hunk 1]
```diff
@@ -182,6 +183,28 @@
...
 
+    <profiles>
+        <profile>
+            <id>owasp-dependency-check</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.owasp</groupId>
+                        <artifactId>dependency-check-maven</artifactId>
+                        <version>5.3.2</version>
```
Comment (ekartsonakisamido) [inline java/pom.xml:194]: Use the `property` defined above.

File: java/pom.xml [hunk 2]
```diff
@@ -35,6 +35,7 @@
         <junit-jupiter.version>5.6.2</junit-jupiter.version>
         <equals-verifier.version>3.4.1</equals-verifier.version>
         <maven-compiler-plugin.version>3.8.1</maven-compiler-plugin.version>
+        <dependency-check-maven.version>5.3.2</dependency-check-maven.version>
```
Comment (ekartsonakisamido) [inline java/pom.xml:38]: Let's use the naming convention, ie `owasp-dependency-check-plugin.version` .

**Review by ekartsonakisamido — CHANGES_REQUESTED** [review]
Minor comments left inline.

**Comment by ekartsonakisamido** [comment]
> > Does this need anything to hook into the build?
> 
> Yes

@ElvenSpellmaker update the build command:
`mvn clean install [...] -Powasp-dependency-check` and
publish the generated reports - @Akrish17 what's the directory & name of the report please ?

**Comment by ElvenSpellmaker** [comment]
@ekartsonakisamido We can't publish the reports as a nice HTML page due to how AZD works, but we can for now attach them as a build artefact for people to download and look at.

> mvn clean install [...] -Powasp-dependency-check
About the command to use to run it, do you know which plug-in and goal it's run under please? We're trying to separate steps out so running whole lifecycles kinda ruins the flow in Azure DevOps.

**Comment by ElvenSpellmaker** [comment]
The `Test` tab and `Code Coverage` tabs are special tabs, and you can't define your own tabs without extensions.

**Comment by ekartsonakisamido** [comment]
> @ekartsonakisamido We can't publish the reports as a nice HTML page due to how AZD works, but we can for now attach them as a build artefact for people to download and look at.
> 
> > mvn clean install [...] -Powasp-dependency-check
> 
> About the command to use to run it, do you know which plug-in and goal it's run under please? We're trying to separate steps out so running whole lifecycles/phases kinda ruins the flow in Azure DevOps.

I'd like to understand the limitation of AZD, it's trivial to publish reports in other CI tools. 

The plugin is `dependency-check:check` and runs on the maven `verify` stage.

**Comment by ElvenSpellmaker** [comment]
Sure, let's have a chat about it. 🙂 

Also, as a slight aside the Code Coverage report is also parsed and any external CSS/JS is stripped, leading me to have to use a `node` container in the build to run some tasks for inlining the resources:
https://github.com/amido/stacks-java/blob/master/build/azDevops/azure/templates/steps/build-java.yml#L102-L111

using the files here: 
https://github.com/amido/stacks-java/tree/master/build/azDevops/azure/coverage

**Comment by Akrish17** [comment]
> > > Does this need anything to hook into the build?
> > 
> > 
> > Yes
> 
> @ElvenSpellmaker update the build command:
> `mvn clean install [...] -Powasp-dependency-check` and
> publish the generated reports - @Akrish17 what's the directory & name of the report please ?

@ekartsonakisamido  the reports are generated in the Target folder and the file generated is called dependency-check-report.html. All these information is added to the task i have created for @ElvenSpellmaker

**Comment by ekartsonakisamido** [comment]
> > > > Does this need anything to hook into the build?
> > > 
> > > 
> > > Yes
> > 
> > 
> > @ElvenSpellmaker update the build command:
> > `mvn clean install [...] -Powasp-dependency-check` and
> > publish the generated reports - @Akrish17 what's the directory & name of the report please ?
> 
> @ekartsonakisamido the reports are generated in the Target folder and the file generated is called dependency-check-report.html. All these information is added to the task i have created for @ElvenSpellmaker

brilliant

**Review by ekartsonakisamido — APPROVED** [review]

**Comment by Akrish17** [comment]
> @ekartsonakisamido We can't publish the reports as a nice HTML page due to how AZD works, but we can for now attach them as a build artefact for people to download and look at.
> 
> > mvn clean install [...] -Powasp-dependency-check
> 
> About the command to use to run it, do you know which plug-in and goal it's run under please? We're trying to separate steps out so running whole lifecycles/phases kinda ruins the flow in Azure DevOps.

so this check is added in maven pom.xml as a profile 
image
This is the command that needs to be used to generate the report -- mvn clean install -Powasp-dependency-check

**Pull request merged**